core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/RecipeEngine.java

```diff
@@ -56,16 +53,16 @@ public void executePreClusterManagerRecipes(Stack stack, Collection<Recipe> reci
         }
     }
 
-    // note: executed when LDAP config is present, because later the LDAP sync is hooked for this salt state in the top.sls.
     public void executePostAmbariStartRecipes(Stack stack, Collection<Recipe> recipes) throws CloudbreakException {
-        if ((stack.getCluster() != null && ldapConfigService.isLdapConfigExistsForEnvironment(stack.getEnvironmentCrn(), stack.getName()))
-                || recipesFound(recipes, POST_CLOUDERA_MANAGER_START)) {
+        if (shouldExecuteRecipeOnStack(recipes, POST_CLOUDERA_MANAGER_START)) {
```

Comment:
is this condition is not valid anymore?
`ldapConfigService.isLdapConfigExistsForEnvironment(stack.getEnvironmentCrn(), stack.getName())`